cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/FileSetProperties.java

```diff
@@ -104,6 +224,75 @@ public Builder setOutputProperty(String name, String value) {
     }
 
     /**
+     * Enable explore for this dataset.
+     */
+    public Builder setExploreEnabled(boolean enabled) {
+      delegate.add(PROPERTY_EXPLORE_ENABLED, Boolean.toString(enabled));
+      return this;
+    }
+
+    /**
+     * Set the class name of the SerDe used to create the Hive table.
+     */
+    public Builder setSerde(String className) {
+      delegate.add(PROPERTY_EXPLORE_SERDE, className);
+      return this;
+    }
+
+    /**
+     * Set the class name of the SerDe used to create the Hive table.
+     */
+    public Builder setSerde(Class<?> serde) {
+      return setSerde(serde.getName());
+    }
+
+    /**
+     * Set the input format used to create the Hive table.
+     * Note that this can be different than the input format used
+     * for the file set itself.
+     */
+    public Builder setExploreInputFormat(String className) {
```

Comment:
we should also have a corresponding method for setInputFormat() that takes a string instead of just the one that takes a class. Same for setOutputFormat